### PR TITLE
[DO NOT MERGE] fmt 4 93

### PR DESCRIPTION
**Pull request by StefanKarpinski**

### CODE REVIEW

**Comment by ararslan** [comment]
Hopefully the style "fixes" are configurable somehow? 😬

**Comment by StefanKarpinski** [comment]
Options are discouraged 😬. The format isn't perfect yet, but it's already pretty good. I think we'll probably start by applying this to Pkg.jl or something smaller than all of Base.

**Comment by ararslan** [comment]
> Options are discouraged

wat

File: base/Base.jl [hunk 1]
```diff
@@ -139,7 +139,7 @@ include("abstractdict.jl")
 
 include("iterators.jl")
 using .Iterators: zip, enumerate
-using .Iterators: Flatten, Filter, product  # for generators
+using .Iterators: Flatten, Filter, product # for generators
```
Comment (JeffBezanson) [inline base/Base.jl]: Why?

File: base/abstractdict.jl [hunk 2]
```diff
@@ -506,101 +510,114 @@ mutable struct IdDict{K,V} <: AbstractDict{K,V}
...
 IdDict(kv::Tuple{}) = IdDict()
 
-IdDict(ps::Pair{K,V}...)           where {K,V} = IdDict{K,V}(ps)
-IdDict(ps::Pair{K}...)             where {K}   = IdDict{K,Any}(ps)
-IdDict(ps::(Pair{K,V} where K)...) where {V}   = IdDict{Any,V}(ps)
-IdDict(ps::Pair...)                            = IdDict{Any,Any}(ps)
+IdDict(ps::Pair{K,V}...) where {K,V} = IdDict{K,V}(ps)
+IdDict(ps::Pair{K}...) where {K} = IdDict{K,Any}(ps)
+IdDict(ps::(Pair{K,V} where K)...) where {V} = IdDict{Any,V}(ps)
+IdDict(ps::Pair...) = IdDict{Any,Any}(ps)
```
Comment (JeffBezanson) [inline base/abstractdict.jl]: :cry:

**Comment by JeffBezanson** [comment]
:scream_cat:

**Comment by StefanKarpinski** [comment]
Folks, chill out. This was an early experiment.

**Comment by quinnj** [comment]
image

File: base/atomics.jl [hunk 3]
```diff
@@ -314,17 +327,23 @@ julia> x[]
 """
 function atomic_min! end
 
-unsafe_convert(::Type{Ptr{T}}, x::Atomic{T}) where {T} = convert(Ptr{T}, pointer_from_objref(x))
+unsafe_convert(::Type{Ptr{T}}, x::Atomic{T}) where {T} =
+    convert(Ptr{T}, pointer_from_objref(x))
 setindex!(x::Atomic{T}, v) where {T} = setindex!(x, convert(T, v))
 
 const llvmtypes = IdDict{Any,String}(
-    Bool => "i8",  # julia represents bools with 8-bits for now. # TODO: is this okay?
```
Comment (vtjnash) [inline base/atomics.jl]: are source comments not allowed anymore?

File: base/bitset.jl [hunk 4]
```diff
@@ -215,24 +230,24 @@ function _matched_map!(f, a1::Bits, b1::Int, a2::Bits, b2::Int,
...
-            # correct value
+        # We don't need to worry about the trailing bits — they already have the
+        # correct value
         end
     end
 
     if bdiff < 0
         if left_false_is_false
-            # We don't need to worry about the leading bits — they're all false
+        # We don't need to worry about the leading bits — they're all false
```
Comment (vtjnash) [inline base/bitset.jl]: huh?

File: base/c.jl [hunk 5]
```diff
@@ -296,34 +294,40 @@ function transcode(::Type{UInt16}, src::AbstractVector{UInt8})
     dst = UInt16[]
     i, n = 1, length(src)
     n > 0 || return dst
-    sizehint!(dst, 2n)
+    sizehint!(dst, 2 * n)
```
Comment (vtjnash) [inline base/c.jl]: lol

File: base/c.jl [hunk 6]
```diff
@@ -296,34 +294,40 @@ function transcode(::Type{UInt16}, src::AbstractVector{UInt8})
...
     a = src[1]
     while true
         if i < n && -64 <= a % Int8 <= -12 # multi-byte character
-            b = src[i += 1]
+            b = src[i+=1]
```
Comment (vtjnash) [inline base/c.jl]: wait, why do other operators get spaces, but `+=` isn't allowed them here?

File: base/cartesian.jl [hunk 7]
```diff
@@ -170,7 +170,7 @@ macro nall(N::Int, criterion::Expr)
     if criterion.head != :->
         throw(ArgumentError("second argument must be an anonymous function expression yielding the criterion"))
     end
-    conds = Any[ Expr(:escape, inlineanonymous(criterion, i)) for i = 1:N ]
+    conds = Any[Expr(:escape, inlineanonymous(criterion, i)) for i = 1:N]
```
Comment (vtjnash) [inline base/cartesian.jl]: @jrevels will be very happy

File: base/atomics.jl (hunk 3 above)
Comment (StefanKarpinski) [reply base/atomics.jl]: it's a bug, yo, and listed in [known issues](https://github.com/domluna/JLFmt.jl#present-limitations)

File: base/c.jl (hunk 6 above)
Comment (domluna) [reply base/c.jl]: afaik most of the time things inside `foo[...]` have no spaces.

File: base/bitset.jl (hunk 4 above)
Comment (domluna) [reply base/bitset.jl]: Yeah it doesn't mess with your comments anymore, was trying to be too fancy before

File: base/Base.jl (hunk 1 above)
Comment (domluna) [reply base/Base.jl]: leaves whitespace after code alone for these kind of comments now

File: base/c.jl (hunk 5 above)
Comment (domluna) [reply base/c.jl]: oooo nice catch, I'll fix that

**Comment by quinnj** [comment]
Great work @domluna, sorry if any comments here felt aggressive or attacking; syntax/code formatting can certainly have a lot of opinions from different people, but it's great to have someone tackling this in Julia.

**Comment by domluna** [comment]
@quinnj not at all, I'm loving the feedback. It can only make the project better for everyone!

**Comment by goggle** [comment]
Is there a place where it can be discussed how the preferred Julia code formatting should look?

**Comment by domluna** [comment]
@goggle Either open an issue https://github.com/domluna/JuliaFormatter.jl/issues or you can discuss on the #parsing slack channel

**Pull request closed without merging**